android/src/main/java/com/onesignal/flutter/OneSignalPlugin.java

```diff
@@ -96,12 +100,29 @@ private void onDetachedFromEngine() {
     OneSignal.setInAppMessageClickHandler(null);
   }
 
+  @Override
+  public void onAttachedToActivity(@NonNull ActivityPluginBinding binding) {
+    this.context = binding.getActivity();
+  }
+
+  @Override
+  public void onDetachedFromActivity() {
+  }
```

Comment:
Should our activity reference be cleared here?

Reply:
Thanks for your input @westy92,
Since we only need to get the activity on initialization and use the activity only for `setAppId`, we don't to track it after.